torch_harmonics/distributed/distributed_sht.py

```diff
@@ -556,42 +459,22 @@ def __init__(self, nlat, nlon, lmax=None, mmax=None, grid="lobatto", norm="ortho
         # determine the dimensions
         self.mmax = mmax or self.nlon // 2 + 1
 
-        # spatial paddings
-        latdist = (self.nlat + self.comm_size_polar - 1) // self.comm_size_polar
-        self.nlatpad = latdist * self.comm_size_polar - self.nlat
-        londist = (self.nlon + self.comm_size_azimuth - 1) // self.comm_size_azimuth
-        self.nlonpad = londist * self.comm_size_azimuth - self.nlon
-
-        # frequency paddings
-        ldist = (self.lmax + self.comm_size_polar - 1) // self.comm_size_polar
-        self.lpad = ldist * self.comm_size_polar - self.lmax
-        mdist = (self.mmax + self.comm_size_azimuth - 1) // self.comm_size_azimuth
-        self.mpad = mdist * self.comm_size_azimuth - self.mmax
+        # compute splits 
+        self.lat_shapes = compute_split_shapes(self.nlat, self.comm_size_polar)
```

Comment:
are those routines general enough to warrant a separate routine?

Reply:
those were changed to be more flexible and it is absolutely crucial that those are in a separate routine. that way, one can always split the tensors consistently across ranks and we do not have to reimplement that for every layer with distributed tensors.

Reply:
this is basically a helper function which should be at the basis of a new distributed framework. it is used in some of the distributed primitives as well and ensures consistency in computing splits. if splits are not computed consistently, then one can easily produce NCCL hangs.